### ISSUE TITLE
How to match backticks in styles?

### ISSUE THREAD

**Issue by abingham** [vale-cli/vale#43]
I think I'm just being dense, but I can't figure out how to match backticks ("`" characters) in a swap rule. I'm linting a book on Python, and we want to standardize how we spell e.g. "for-loop". So I want a rule like this (simplified from what we really want):
```yaml
swap:
  'for loop': for-loop
  '`for` loop': for-loop
```
But this doesn't seem to work. The first rule works just fine, but the second one with backticks doesn't seem to fire. How can I make this work?

**Comment by abingham** [comment]
Does vale actually just ignore the backticks, quotation marks, and other "surrounding stuff"?

**Comment by jdkato** [comment]
By default, Vale ignores the syntax equivalent of `<pre>` and `<code>` tags in markup and everything but comments in source code. You can disable this behavior by passing `--ignore-syntax`, but you'll then also lose the ability to target things like headings and lists individually. 

For example, given a Markdown file with the following contents:

```text
This is a for loop. This is also a `for` loop.
```
and this rule definition:

```yaml
# Named Back.yml
extends: substitution
message: Consider using '%s' instead of '%s'
level: warning
nonword: true # "`" is a non-word character
swap:
  'for loop': for-loop
  '`for` loop': for-loop
```

Running `vale --output=line test.md` yields:

```
test.md:1:11:test.Back:Consider using 'for-loop' instead of 'for loop'
```

And `vale --ignore-syntax --output=line test.md` gives the desired:

```
test.md:1:11:test.Back:Consider using 'for-loop' instead of 'for loop'
test.md:1:36:test.Back:Consider using 'for-loop' instead of '`for` loop'
```

---

In a future build, I'd like to add an `IgnoredScopes` setting to the config file to make this a little more flexible. I also should expand on this more in the docs probably.

**Comment by abingham** [comment]
OK, thanks for the details. That works well for me. Should I close this out, or do you want to keep it open?

**Comment by jdkato** [comment]
I'll leave it open until I get around to improving the documentation on the command line flags.

**Comment by jdkato** [comment]
I've added a new option to make code spans available on a per-rule basis (see #44). So, the updated definition would be:

```yaml
extends: substitution
message: Consider using '%s' instead of '%s'
level: warning
nonword: true
code: true
swap:
  'for loop': for-loop
  '`for` loop': for-loop
```
And you can run normally (without `--ignore-syntax`).

(Updated docs are also forthcoming.)

**Comment by abingham** [comment]
Brilliant, thanks!
